File: week10/week10-1.cpp
/// week10-1.cpp 函式呼叫函式 Recursion 遞迴 N階層
/// 以前是用 for迴圈寫, 今天用「函式呼叫函式」來寫
#include <stdio.h>
int func(int n) {
    if(n==1) return 1; ///終止條件, 像「數學歸納法」N=1成立
    return n * func(n-1); /// 函式呼叫函式
}
int main()
{
    printf("請輸入N:");
    int N;
    scanf("%d", &N);
    int ans = func(N);
    printf("%d", ans);
}

File: week10/week10-1.py
# week10-1.py  函式呼叫函式 Recursion 遞迴 N階層
N = int(input('請輸入N:'))
# 以前是用 for迴圈寫, 今天用「函式呼叫函式」來寫
def func(n):
  if n==1: return 1 # 終止條件, 像「數學歸納法」N=1成立
  return n * func(n-1) # 函式呼叫函式, 把大問題, 改成去問「小問題」
ans = func(N)
print(ans)